src/main/scala/com/twitter/finagle/postgres/Client.scala

```diff
@@ -81,6 +81,8 @@ class Client(factory: ServiceFactory[PgRequest, PgResponse], id:String) {
       service =>
         parse(sql, Some(service)).map { name =>
           new PreparedStatementImpl(name, service)
+        }.onFailure {
+          err => service.close()
```

Comment:
You can't close it at this point if it's a success, because the service still has more phases to execute after Parse.